### DISCUSSION TITLE
How to access store from service class

### DISCUSSION THREAD

**Discussion by vionijannet** [nuxt/nuxt#19264] [Questions]
I'm using Pinia as state management of nuxt. 
This is the store I made. If there is something not quite right about the store, please let me know.

```
// store/userStore.ts
import { defineStore } from "pinia";
export const useUserStore = defineStore("userStore", () => {
    const token = ref("");

    function setToken(newToken: string): void {
        token.value = newToken;
    }

    function removeToken(): void {
        token.value = "";
    }

    return { token, setToken, removeToken };
})
```

I also create a class to hit service 
```
// service/BaseService.ts
export class BaseService {
    readonly BASE_API_ENDPOINT = "/api";

    constructor(
        protected userStore: Store<"userStore">
    ) {}
    
    httpGet(
        url: string,
        config: any
    ): any {
        // I need the token in here
        this.userStore.?
    }
}
```

I tried to put the store as parameter of class, but I still have no idea how to get the token from the store. 
Anyone know how to access token from the store?

**Comment by ndragun92** [comment]
Did not try to access it that way but you can write

```typescript
import useUserStore from '@/store/userStore'

const userStore = useUserStore()

userStore.setToken('token')
```

If you cannot use it in such service instead of creating service like that I would move logic in composables

```typescript
import useUserStore from '@/store/userStore'
export default function useBaseService() {
   const BASE_API_ENDPOINT = "/api";

  const httpGet = (url: string, config: any) => {
      const userStore = useUserStore()
      userStore.setToken('token')
   }
}
```

**Reply by vionijannet** [reply]
> Did not try to access it that way but you can write
> 
> ```ts
> import useUserStore from '@/store/userStore'
> 
> const userStore = useUserStore()
> 
> userStore.setToken('token')
> ```
> 
I tried this one and it works, but I prefer to access store as a parameter for testing purpose. 
And I cannot change BaseService to function due to the project's structure.
